src/mtl.rs

```diff
@@ -79,50 +111,52 @@ impl<'a> From<Material> for Cow<'a, Material> {
 struct Parser<I>(I);
 
 impl<'a, I: Iterator<Item = &'a str>> Parser<I> {
-    fn get_vec(&mut self) -> Option<[f32; 3]> {
+    fn get_vec(mut self) -> Result<[f32; 3], MtlError> {
```

Comment:
why are all the function signatures changed to accept `self` by value?

Reply:
Good catch. This change stemmed from my use of fold inside `get_string` as fold takes the iterator (`self.0`) by value which caused issues. I think this behavior is reasonable for get_string as it consumes the rest of the line (and the iterator). I have changed the rest back to `&mut self` however as they don't consume the whole line.